pkg/controller/bootstrap/tokencleaner.go

```diff
@@ -149,7 +149,11 @@ func (tc *TokenCleaner) processNextWorkItem() bool {
 	}
 	defer tc.queue.Done(key)
 
-	if err := tc.syncFunc(key.(string)); err != nil {
+	if secret, ok := key.(*v1.Secret); ok {
```

Comment:
this is not correct. only strings should be added to the queue

Reply:
this line should be fixed (the evalSecret method should take the key and requeue it, not enqueue the secret):
```
tc.queue.AddAfter(o, ttl)
```

we also need a test exercising processing the queue after enqueueing because of a ttl

cc @zjj2wry

Reply:
evalSecret checks whether the secret has expired. If not, it enqueues it with ttl.

Reply:
correct, but it should enqueue the key, not the secret